src/ekka_cluster_etcd.erl

```diff
@@ -167,3 +227,148 @@ rand_addr([Addr]) ->
 rand_addr(AddrList) ->
     lists:nth(rand:uniform(length(AddrList)), AddrList).
 
+%%--------------------------------------------------------------------
+%% v3
+%%--------------------------------------------------------------------
+etcd_v3(Action) ->
+    gen_server:call(?SERVER, Action, 5000).
+
+v3_discover(#state{prefix = Prefix}) ->
+    Context = v3_nodes_context(Prefix),
+    case eetcd_kv:get(Context) of
+        {ok, Response} ->
+            case maps:get(kvs, Response) of
+                [] ->
+                    {ok, []};
+                KvsList ->
+                    Nodes = [
+                        binary_to_atom(maps:get(value, Kvs), utf8) || Kvs <- KvsList],
+                    {ok, Nodes}
+            end;
+        Error ->
+            Error
+    end.
+
+v3_lock(#state{prefix = Prefix, lease_id = ID}) ->
+    case eetcd_lock:lock(?MODULE, list_to_binary(v3_lock_key(Prefix)), ID) of
+        {ok, #{key := LockKey}} ->
+            persistent_term:put(ekka_cluster_etcd_lock_key, LockKey),
+            ok;
+        Error ->
+            Error
+    end.
+
+v3_unlock(_) ->
+    case persistent_term:get(ekka_cluster_etcd_lock_key, undefined) of
+        undefined ->
+            {error, lock_lose};
+        LockKey ->
+            case eetcd_lock:unlock(?MODULE, LockKey) of
+                {ok, _} ->
+                    persistent_term:erase(ekka_cluster_etcd_lock_key),
+                    ok;
+                Error ->
+                    Error
+            end
+    end.
+
+v3_register(#state{prefix = Prefix ,lease_id = ID}) ->
+    Context = v3_node_context(Prefix, ID),
+    case eetcd_kv:put(Context) of
+        {ok, _Response} ->
+            ok;
+        Error ->
+            Error
+    end.
+
+v3_unregister(#state{prefix = Prefix}) ->
+    Context = v3_node_context_only_key(Prefix),
```

Comment:
Unregister key and register key are inconsistent

Reply:
Register context is
``` erlang
 #{key => Prefix ++ "/ekkacl/nodes/" ++ atom_to_list(node()) , value => atom_to_list(node())} 
```
and unregister only need key
``` erlang
#{key => Prefix ++ "/ekkacl/nodes/" ++ atom_to_list(node())}
```